tests/integration-tests/cfn_stacks_factory.py

```diff
@@ -92,6 +101,15 @@ def get_private_subnet(self):  # TODO add possibility to override default
         assert_that(az_id_tag).is_not_none()
         return self.cfn_outputs[f"{az_id_tag}PrivateSubnetId"]
 
+    def get_all_private_subnets(self):
```

Comment:
Probably better to have one method that implements the logic 

    def _get_all_subnets(self, visibility="Public"):
        assert_that(self.az_ids).is_not_none()
        subnet_ids = []
        for az_id in self.az_ids:
            az_id_tag = to_pascal_from_kebab_case(az_id)
            subnet_ids.append(self.cfn_outputs[f"{az_id_tag}{visibility}SubnetId"])

        return random.shuffle(subnet_ids)

and then have 

    def get_all_public_subnets(self):
        return _get_all_subnets("Public")
    
  
    def get_all_public_subnets(self):
        return _get_all_subnets("Private")


and likely the same holds for  `get_public_subnet` and `get_private_subnet`